drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -1677,20 +1688,21 @@ static int bcm2835_isp_remove(struct platform_device *pdev)
 					      dev->component);
 
 	vchiq_mmal_finalise(dev->mmal_instance);
-
-	return 0;
 }
 
-static int bcm2835_isp_probe(struct platform_device *pdev)
+static int bcm2835_isp_probe_instance(struct platform_device *pdev,
+				      struct bcm2835_isp_dev **dev_int,
+				      unsigned int instance)
 {
 	struct bcm2835_isp_dev *dev;
 	unsigned int i;
 	int ret;
 
-	dev = devm_kzalloc(&pdev->dev, sizeof(*dev), GFP_KERNEL);
-	if (!dev)
+	*dev_int = devm_kzalloc(&pdev->dev, sizeof(*dev), GFP_KERNEL);
+	if (!*dev_int)
 		return -ENOMEM;
 
+	dev = *dev_int;
```

Comment:
Can we allocate and store as ```dev``` initially, and only assign to ```*dev_int``` as the last thing on success? Then we don't have a stale non-NULL pointer kicking around in the global state.

Reply:
Ack.